h2o-docs/src/product/data-science/gam.rst

```diff
@@ -401,6 +401,114 @@ For documentation on thin plate regression splines, refer here:
+
+.. math::
+   
+   N+1+k-2
+
+where:
+
+- :math:`N+1` is the number of knots over the input range without duplication
+- :math:`k` is the order of the spline
+
+**M-splines:** :math:`M_{i,k}(t)`
+
+If you normalize the basic B-spline function to have an integration of 1 over the interest range where it is non-zero, you can denote it as :math:`M_{i,k}(t)`. This is the normalized B-spline Type I, and it is defined as:
+
+.. math::
+   
+   M_{i,k}(t) = k \times Q_{i,k}(t)
+
+You can also derive :math:`M_{i,k}(t)` using the following recursive formula:
+
+.. math::
+   
+   M_{i,k}(t) = {\frac{k}{k-1}}\bigg( {\frac{(t-t_i)}{(t_{i+k}-t_i)}}M_{i,k-1}(t)-{\frac{(t_{i+k}-t)}{(t_{i+k}-t_i)}}M_{i+1,k-1}(t)\bigg)
+
+Note that :math:`M_{i,k}(t)` is defined over the same knot sequence as the original B-spline, and the number of :math:`M_{i,k}(t)` splines is the same as the number of B-splines over the same known sequence.
+
+**N-splines:** :math:`N_{i,k}(t)`
+
+The N-splines are normalized to have a summation of 1 when :math:`t_0 \leq t < t_N` as :math:`\sum_{i=0}^{N+k-1}N_{i,k}(t) = 1`. :math:`N_{i,k}(t)` is the normalized B-spline Type II in this implementation. The N-splines share the same knot sequence with the original M-spline and B-spline. The N-spline can be derived from the M-spline or the B-spline using:
+
+.. math::
+   
+   N_{i,k}(t) = {\frac{(t_{i+k}-t_i)}{k}}M_{i,k}(t) = (t_{i+k}-t_i)Q_{i,k}(t)
+
+Or, you can use the recursive formula where higher order N-splines can be derived from two lower order N-splines:
+
+.. math::
+   
+   N_{i,k}(t) = {\frac{t-t_i}{t_{i+k-1}-t_i}}N_{i,k-1}(t)+{\frac{t_{i+k}-t}{t_{i+k}-t_{i+1}}}N_{i+1,k-1}(t) 
+
+**I-splines:** :math:`I_{i,k}(t)`
+
+I-splines are used to build the monotone spline functions by restricting the gamified column coefficients to be :math:`\geq` 0 [:ref:`5<ref5>`]. We have implemented I-splines using the following method:
+
+.. math::
+   
+   I_{i,k}(t) = \sum_{l=1}^{i+r}N_{l,k+1}(t), t \leq t_{i+r+1}
+
+**Penalty Matrix for I-splines**
+
+The objective function used to derive the coefficients for regression is:
+
+.. math::
+   
+   \sum_{i=0}^n \Bigg(y_i- \bigg(\sum_{j=0}^{numBasis-1}I_{j,k}(t_i)\beta_j + \beta_0 \bigg)\Bigg)^2 + \lambda\beta^T penaltyMat\beta
+
+The second derivative of all basis functions is defined as:
+
+.. math::
+   
+   I_k^{2ndDeriv} = \begin{bmatrix} {\frac{d^2(I_{0,k}(t))}{d^2t}} \\
+   {\frac{d^2(I_{1,k}(t))}{d^2t}} \\ \vdots \\
+   {\frac{d^2(I_{numBasis-2,k}(t))}{d^2t}} \\
+   {\frac{d^2(I_{numBasis-1,k}(t))}{d^2t}} \\\end{bmatrix}
+
+where the penalty matrix (:math:`penaltyMat`) for I-spline :math:`I_{j,k}(t)` is defined as:
+
+.. math::
+   
+   penaltyMat = \int_{t_0}^{t_N}I_k^{2ndDeriv} \times \text{ transpose of }(I_k^{2ndDeriv})dt
+
+Element at row :math:`m` and column :math:`n` of :math:`penaltyMat` is
+
+.. math::
+   
+   penaltyMat_{m,n} = \int_{t_0}^{t_N}{\frac{d^2(I_{m,k}(t))}{d^2t}}{\frac{d^2(I_{n,k}(t))}{d^2t}}dt
```

Comment:
Hannah:  Not sure where you can add this but please let the user know that they have to specify monotone spline by specifying bs=2 and then specify the spline_orders parameter.

Reply:
Added in the opening